src/restore/restore_workspace_creator.py

```diff
@@ -36,9 +39,11 @@ def __create_target_dataset(self, source_table_reference,
             table_expiration
         )
 
-    def __create_empty_partitioned_table_if_not_exists(self, target_table_reference):
-        if target_table_reference.is_partition() and not self.BQ.get_table_by_reference_cached(target_table_reference).table_exists():
-            self.BQ.create_empty_partitioned_table(target_table_reference)
+    def __create_empty_partitioned_table_if_not_exists(self, source_table_reference, target_table_reference):
+        target_table_metadata = BigQueryTableMetadata.get_table_by_reference_cached(target_table_reference)
+        if target_table_reference.is_partition() and not target_table_metadata.table_exists():
+            source_table_metadata = BigQueryTableMetadata.get_table_by_reference_cached(source_table_reference)
```

Comment:
What in case the source data is deleted? :) You need to copy metadata from one of the backups.

Reply:
As discussed source_table_metadata in this case is particular backup metadata, so everything is fine